rust/connlib/tunnel/src/device_channel/tun_windows.rs

```diff
@@ -41,7 +51,45 @@ impl Tun {
             }
         };
 
-        adapter.set_address(config.ipv4)?;
+        // TODO: I think wintun flashes a couple console windows here when it shells out to netsh. We should upstream the same patch I'm doing for powershell to the wintun project
+        adapter.set_network_addresses_tuple(
+            config.ipv4.into(),
+            [255, 255, 255, 255].into(),
+            None,
+        )?;
+        adapter.set_network_addresses_tuple(
+            config.ipv6.into(),
+            Ipv6Addr::new(
+                0xffff, 0xffff, 0xffff, 0xffff, 0xffff, 0xffff, 0xffff, 0xffff,
+            )
+            .into(),
+            None,
+        )?;
+
+        let iface_idx = adapter.get_adapter_index()?;
+        tracing::debug!("adapter index = {}", iface_idx);
+
+        // Remove any routes that were previously associated with us
+        // TODO: Pick a more elegant way to do this
+        Command::new("powershell")
+            .creation_flags(CREATE_NO_WINDOW)
+            .arg("-Command")
+            .arg(format!(
+                "Remove-NetRoute -InterfaceIndex {iface_idx} -Confirm:$false"
+            ))
+            .stdout(Stdio::null())
+            .status()?;
+
+        // Set our DNS IP as the DNS server for our interface
+        // TODO: Lots of issues with this. Windows does seem to use it, but I'm not sure why. And there's a delay before some Firefox windows pick it up. Curl might be picking it up faster because its DNS cache starts cold every time.
+        Command::new("powershell")
+            .creation_flags(CREATE_NO_WINDOW)
+            .arg("-Command")
+            .arg(format!(
+                "Set-DnsClientServerAddress -InterfaceIndex {iface_idx} -ServerAddresses(\"100.100.111.1\")"
+            ))
+            .stdout(Stdio::null())
+            .status()?;
```

Comment:
See the pain we experienced doing something similar in Apple:

https://github.com/firezone/firezone/pull/2947

Unfortunately I don't have any tips to offer to avoid similar pain on Windows, but the main thing we want to prevent is DNS routing loops where we set our sentinel as the only resolver, then try to read it to use for fallback. What I did on Apple was _also_ set the previous system resolvers and secondaries, so that they're available from reading `/etc/resolv.conf` later on.